### DISCUSSION TITLE
Request for letting me know adjust the timeout settings on Hosted Budibase app

### DISCUSSION THREAD

**Discussion by quantsegu** [Budibase/budibase#12805] [Help]
Hi

I'm receiving "timeout" for REST POST Calls to ChatGPT, so i want to extend time for response. Can any one help?

Thanks!

![image](https://github.com/Budibase/budibase/assets/1640267/812ba7ea-83df-4894-a411-449b8520e8e2)

**Comment by melohagan** [comment]
Hey @quantsegu 

10 seconds is a long time to be waiting on a REST request, perhaps it would be better to do this asynchronously with automations?

**Reply by quantsegu** [reply]
How can i do that? can you please share any information @melohagan

**Reply by melohagan** [reply]
What is your use-case for ChatGPT?

**Reply by quantsegu** [reply]
@melohagan  i want to call chatgpt for a response based on user input, where the ChatGPT is Prompt-Engineered for a specific scenario. I am building several scenarios.

I see that the results in this case took 25 seconds. 

Thanks for your help and attention

**Comment by DeveloperMDCM** [comment]
I debugged the Budibase code and found that these variables are necessary to increase the timeout.
https://github.com/DeveloperMDCM/budibase-timeout-fix
